crates/shared/src/signature_validator/simulation.rs

```diff
@@ -106,51 +96,39 @@ impl SignatureValidating for Validator {
 
 #[derive(Debug)]
 struct Simulation {
-    result: SimulationResult,
-    verification: Verification,
-}
-
-#[derive(Debug)]
-struct Verification {
     gas_used: U256,
 }
 
-#[derive(Debug)]
-enum SimulationResult {
-    Ok,
-    Invalid,
-    Revert,
-}
-
 impl Simulation {
     fn decode(output: &[u8]) -> Result<Self> {
         let function = contracts::support::Signatures::raw_contract()
             .abi
             .function("validate")
             .unwrap();
         let tokens = function.decode_output(output).context("decode")?;
-        let (result, gas_used) = Tokenize::from_token(Token::Tuple(tokens))?;
+        let (gas_used,) = Tokenize::from_token(Token::Tuple(tokens))?;
 
-        Ok(Self {
-            result: SimulationResult::decode(result)?,
-            verification: Verification { gas_used },
-        })
+        Ok(Self { gas_used })
     }
 }
 
-impl SimulationResult {
-    fn decode(value: u8) -> Result<Self> {
-        match value {
-            0 => Ok(Self::Ok),
-            1 => Ok(Self::Invalid),
-            2 => Ok(Self::Revert),
-            _ => anyhow::bail!("invalid simulation result variant {value}"),
+impl From<web3::Error> for SignatureValidationError {
+    fn from(err: web3::Error) -> Self {
+        // TODO: This is needed to parse Hardhat revert errors, which
```

Comment:
Fixed in https://github.com/cowprotocol/ethcontract-rs/pull/962

Will remove this extra branch when the new `ethcontract` version hits.

Reply:
Probably worth creating issue so we don't forget.

Reply:
I already made the PR and plan on doing it today. Can create an issue if you think its worth it.

Reply:
https://github.com/cowprotocol/services/issues/1875